src/stores/business.ts

```diff
@@ -213,6 +214,270 @@ export const useBcrosBusiness = defineStore('bcros/business', () => {
+        const ff = !!getFeatureFlag('supported-put-back-on-entities')?.includes(legalType)
+        return (ff && isAllowedToFile(FilingTypes.PUT_BACK_ON))
+      }
+
+      case AllowableActionE.RECORD_CONVERSION: {
+        return isAllowedToFile(FilingTypes.CONVERSION)
+      }
+
+      case AllowableActionE.REGISTRARS_NOTATION: {
+        return isAllowedToFile(FilingTypes.REGISTRARS_NOTATION)
+      }
+
+      case AllowableActionE.REGISTRARS_ORDER: {
+        return isAllowedToFile(FilingTypes.REGISTRARS_ORDER)
+      }
+
+      case AllowableActionE.RESTORATION: {
+        // NB: specific entities are targeted via LaunchDarkly
+        // NB: this applies to full restoration or limited restoration
+        // but not limited restoration extension or limited restoration to full
+        const ff = !!getFeatureFlag('supported-restoration-entities')?.includes(legalType)
+        return (
+          ff &&
+          (
+            isAllowedToFile(FilingTypes.RESTORATION, FilingSubTypeE.FULL_RESTORATION) ||
+            isAllowedToFile(FilingTypes.RESTORATION, FilingSubTypeE.LIMITED_RESTORATION)
+          )
+        )
+      }
+
+      case AllowableActionE.STAFF_COMMENT: {
+        return (isBusiness && currentAccount.accountType === AccountTypeE.STAFF)
+      }
+
+      case AllowableActionE.TRANSITION: {
+        return isAllowedToFile(FilingTypes.TRANSITION)
+      }
+
+      case AllowableActionE.VOLUNTARY_DISSOLUTION: {
+        // NB: specific entities are targeted via LaunchDarkly
+        const ff = !!getFeatureFlag('supported-dissolution-entities')?.includes(legalType)
+        return (ff && isAllowedToFile(FilingTypes.DISSOLUTION, FilingSubTypeE.DISSOLUTION_VOLUNTARY))
+      }
+
+      default: return false // should never happen
+    }
+  }
+
+  /** Whether the entity is a Cooperative Assocation. */
+  function isEntityCoop (): boolean {
+    return (currentBusiness.value.legalType === CorpTypeCd.COOP)
+  }
+
+  /** Whether the entity is a General Partnership. */
+  function isEntityPartnership (): boolean {
+    return (currentBusiness.value.legalType === CorpTypeCd.PARTNERSHIP)
+  }
+
+  /** Whether the entity is a Sole Proprietorship. */
+  function isEntitySoleProp (): boolean {
+    return (currentBusiness.value.legalType === CorpTypeCd.SOLE_PROP)
+  }
+
+  /** Whether the entity is a Sole Proprietorship or General Partnership. */
+  function isEntityFirm (): boolean {
+    return (isEntitySoleProp() || isEntityPartnership())
+  }
+
+  /** Whether the entity is a BC Limited Company. */
+  function isEntityBcCompany (): boolean {
+    return currentBusiness.value.legalType === CorpTypeCd.BC_COMPANY
+  }
+
+  /** Whether the entity is a Benefit Company. */
+  function isEntityBenefitCompany (): boolean {
+    return currentBusiness.value.legalType === CorpTypeCd.BENEFIT_COMPANY
+  }
+
+  /** Whether the entity is a BC Community Contribution Company. */
+  function isEntityBcCcc (): boolean {
```

Comment:
Could we replace all these 'is...' with one function where you pass the desired legal type in? Otherwise we will end up with 100 of these and I don't think having specific ones adds very much value

Reply:
and you could make the param a list so that we could check if its in a group